modules/gapi/src/backends/cpu/gcpuimgproc.cpp

```diff
@@ -180,6 +180,26 @@ GAPI_OCV_KERNEL(GCPUCanny, cv::gapi::imgproc::GCanny)
     }
 };
 
+GAPI_OCV_KERNEL(GCPUCalcOptFlowLK, cv::gapi::imgproc::GCalcOptFlowLK)
+{
+    static void run(const cv::Mat& prevImg, const cv::Mat& nextImg, const std::vector<cv::Point2f>& prevPts, const std::vector<cv::Point2f>& predPts, const cv::Size& winSize, int maxLevel, const cv::TermCriteria& criteria, int flags, double minEigThresh, std::vector<cv::Point2f>& outPts, std::vector<uchar>& status, std::vector<double>& err)
+    {
+        if (flags & cv::OPTFLOW_USE_INITIAL_FLOW)
+            outPts = std::vector<cv::Point2f>(predPts);
+        cv::calcOpticalFlowPyrLK(prevImg, nextImg, prevPts, outPts, status, err, winSize, maxLevel, criteria, flags, minEigThresh);
```

Comment:
It is planned that outPts will be null-initialized if the condition isn't passed. Should this be done explicitly or outputs from kernels are initialized by nulls automatically?

Reply:
I think that outputs from kernels are not initialized with nulls automatically and you should initialized them explicitly.

Reply:
@TolyaTalamanov , please correct me if I am wrong

Reply:
As all the output vectors are initialized inside the function cv::calcOpticalFlowPyrLK(), it is not necessary here